opentracing-jms-2/src/main/java/io/opentracing/contrib/jms2/TracingJMSProducer.java

```diff
@@ -0,0 +1,433 @@
+    }
+
+    @Override
+    public long getDeliveryDelay() {
+        return jmsProducer.getDeliveryDelay();
+    }
+
+    @Override
+    public int getDeliveryMode() {
+        return jmsProducer.getDeliveryMode();
+    }
+
+    @Override
+    public boolean getDisableMessageID() {
+        return jmsProducer.getDisableMessageID();
+    }
+
+    @Override
+    public boolean getDisableMessageTimestamp() {
+        return jmsProducer.getDisableMessageTimestamp();
+    }
+
+    @Override
+    public double getDoubleProperty(String arg0) {
+        return jmsProducer.getDoubleProperty(arg0);
+    }
+
+    @Override
+    public float getFloatProperty(String arg0) {
+        return jmsProducer.getFloatProperty(arg0);
+    }
+
+    @Override
+    public int getIntProperty(String arg0) {
+        return jmsProducer.getIntProperty(arg0);
+    }
+
+    @Override
+    public String getJMSCorrelationID() {
+        return jmsProducer.getJMSCorrelationID();
+    }
+
+    @Override
+    public byte[] getJMSCorrelationIDAsBytes() {
+        return jmsProducer.getJMSCorrelationIDAsBytes();
+    }
+
+    @Override
+    public Destination getJMSReplyTo() {
+        return jmsProducer.getJMSReplyTo();
+    }
+
+    @Override
+    public String getJMSType() {
+        return jmsProducer.getJMSType();
+    }
+
+    @Override
+    public long getLongProperty(String arg0) {
+        return jmsProducer.getLongProperty(arg0);
+    }
+
+    @Override
+    public Object getObjectProperty(String arg0) {
+        return jmsProducer.getObjectProperty(arg0);
+    }
+
+    @Override
+    public int getPriority() {
+        return jmsProducer.getPriority();
+    }
+
+    @Override
+    public Set<String> getPropertyNames() {
+        return jmsProducer.getPropertyNames();
+    }
+
+    @Override
+    public short getShortProperty(String arg0) {
+        return jmsProducer.getShortProperty(arg0);
+    }
+
+    @Override
+    public String getStringProperty(String arg0) {
+        return jmsProducer.getStringProperty(arg0);
+    }
+
+    @Override
+    public long getTimeToLive() {
+        return jmsProducer.getTimeToLive();
+    }
+
+    @Override
+    public boolean propertyExists(String arg0) {
+        return jmsProducer.propertyExists(arg0);
+    }
+
+    @Override
+    public JMSProducer send(Destination destination, Message message) {
+        Span span = TracingMessageUtils.buildAndInjectSpan(destination, message, tracer);
+        try {
+            jmsProducer.send(destination, message);
+        } catch (Throwable e) {
+            SpanJmsDecorator.onError(e, span);
+            throw e;
+        } finally {
+            span.finish();
+        }
+        return this;
+    }
+
+    @Override
+    public JMSProducer send(Destination destination, String message) {
+        TextMessage textMsg = null;
+        try {
+            textMsg = getTextMessage();
+            textMsg.setText(message);
+        } catch (JMSException e) {
+            e.printStackTrace();
```

Comment:
if we catch exception it means `testMsg` can be null, so later we will get NPE. 
I propose call original `jmsProducer.send(destination, message)` and return.

Reply:
Without call TracingMessageUtils.buildAndInjectSpan() ??
Juste like:
```java
@Override
    public JMSProducer send(Destination destination, String message) {
        jmsProducer.send(destination, message);
        return this;
    }
```
Because in all method I convert the type to Message to use your buildAndInjectSpan().

Reply:
yes, if conversion failed.